apps/workflow-server/gcc-workflow-server-facade/gcc-restclient-facade-default/src/test/java/com/coremedia/labs/translation/gcc/facade/def/GccCredentialsExtension.java

```diff
@@ -35,32 +36,42 @@ public class GccCredentialsExtension implements ExecutionCondition, ParameterRes
   private static final Logger LOG = getLogger(lookup().lookupClass());
 
   private static final String USER_HOME = System.getProperty("user.home");
-  private static final Path GCC_CREDENTIALS_PATH = Paths.get(USER_HOME, ".gcc.properties");
-  private @Nullable Map<String, String> gccProperties;
+  private static final String GCC_PROPERTIES_DEFAULT = ".gcc.properties";
+  private static final String GCC_PROPERTIES_PROFILE_PATTERN = ".gcc.%s.properties";
+  private static final Path GCC_CREDENTIALS_PATH = Paths.get(USER_HOME, GCC_PROPERTIES_DEFAULT);
+  private volatile @Nullable Map<String, String> gccProperties;
```

Comment:
Why volatile? Looks as if all access is already guarded with `synchronized (this)`